tests/chainer_tests/test_gradient_check.py

```diff
@@ -397,5 +397,23 @@ def test_multiple_input_output_cpu(self):
     def test_multiple_input_output_gpu(self):
         self.check_multiple_input_output(cuda.cupy)
 
+    def check_double_backward_with_params(self, xp):
+        arrays = xp.ones((5, 1), dtype='f')
+        x, gy, ggx, param_a, ggparam = arrays
+
+        param = chainer.Variable(param_a)
+
+        def f(x):
+            return x * param
```

Comment:
I think this function is too simple to test 2nd order gradient. How about more complex configuration?

Reply:
Why shouldn't it be simple?

Reply:
2nd order gradient of this function is zero. If `check_double_backward` had some glitch which incorrectly considers 2nd gradient of any function to be zero, this test case would not detect that glitch.

Reply:
No, the second order gradient of `x * param` is not zero, because it is a bilinear function. Taking gradient w.r.t. `x` after taking gradient w.r.t. `param` results in the gradient `1`. Taking gradient w.r.t. `x` twice results in the gradient `0`. Taking gradient w.r.t. `param` after taking gradient w.r.t. `x` results in the gradient `0`. I think this configuration is enough to check if the gradient check correctly handles the parameter.

Reply:
Note that `1` actually depends on the random inputs `y_grad` and `x_grad_grad` so simply setting gradients to `1` does not pass the test.

Reply:
OK.